lib/db/mysql.js

```diff
@@ -566,21 +566,31 @@ module.exports = function (log, error) {
 
   // Internal
 
-  MySql.prototype.singleQuery = function (poolName, sql, params) {
+  MySql.prototype.singleQuery = function (poolName, sql, params, readUncommitted) {
     return this.getConnection(poolName)
       .then(
         function (connection) {
-          return query(connection, sql, params)
-            .then(
-              function (result) {
-                connection.release()
-                return result
-              },
-              function (err) {
-                connection.release()
-                throw err
-              }
-            )
+          if (readUncommitted) {
+            return query(connection, 'SET TRANSACTION ISOLATION LEVEL READ UNCOMMITTED')
```

Comment:
(although I recall the docs being somewhat less than clear about how long-lived the effects of this statement are in various contexts)